lms/services/blackboard_api/client.py

```diff
@@ -186,3 +160,84 @@ def course_groups(
                 raise ExternalAsyncRequestError(response=response)
 
         return self_enrollment_groups + instructor_only_groups
+
+    def _list_files_url(self, course_id, folder_id=None) -> str:
+        """Return the url for the list files API endpoint."""
+        path = f"courses/uuid:{course_id}/resources"
+
+        if folder_id:
+            # Get the files and folders in the given folder instead of the
+            # course's top-level files and folders.
+            path += f"/{folder_id}/children"
+
+        return (
+            path
+            + "?"
+            + urlencode(
+                {
+                    "limit": PAGINATION_LIMIT,
+                    "fields": "id,name,type,modified,mimeType,size,parentId",
+                }
+            )
+        )
+
+    def _store_files(self, course_id, files: List[dict]):
+        """Store the given files in the DB."""
+        self._file_service.upsert(
+            [
+                {
+                    "type": "blackboard_file"
+                    if file["type"] == "File"
+                    else "blackboard_folder",
+                    "course_id": course_id,
+                    "lms_id": file["id"],
+                    "name": file["name"],
+                    "size": file["size"],
+                    "parent_lms_id": file["parentId"],
+                }
+                for file in files
+            ]
+        )
```

Comment:
This is just a call that was already here that's been moved out into a helper method